### PR TITLE
For #20596 remove startup timeline probes

### LINKED ISSUE #20596: Remove StartupTimeline & start up subsection duration probes
While reviewing telemetry probes for renewal https://github.com/mozilla-mobile/fenix/issues/20520, we decided not to renew:
```
Startup.timeline:
  framework_primary
  framework_secondary
  framework_start_error
  framework_start_read_error
  clock_ticks_per_second_v2

perf.startup:
 app_on_create_to_glean_init
 app_on_create_to_megazord_init
 app_on_create_to_setup_in_main
 home_activity_on_create
 home_activity_on_start
 home_fragment_on_create_view
 home_fragment_on_view_created
 base_bfragment_on_create_view
 base_bfragment_on_view_created
```

Because we're not actively using them and have no intention to do so in the near future. We should remove this code.

### PR DESCRIPTION
**Pull request by RotBolt**
### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
  This PR involves probes for telemetry which did not have tests. So did not add any tests
  EDIT: Probes had tests, but since probes have been removed, so removing irrelevant tests as well
  
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
  No UI changes
  
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.
 No accessibility related code
 
### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Closes #20596 

This is the second installment of PR for issue #20596 
I have removed `startup.timeline` probes and their usage from the application code. Also removed the associated functions, files which are no longer used after removal of `startup.timeline` probes

### CODE REVIEW

**Comment by RotBolt** [comment]
cc @mcomella

**Comment by RotBolt** [comment]
There are some code to be removed from tests. Will remove it in some time

File: app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt:22]: chore (non-blocking): afaict, the Stat class is now unused: can we remove it and its test?

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt:80]: thought (non-blocking): this code is scheduling and submitting a telemetry ping: `Pings.startupTimeline`. A telemetry ping is a group of telemetry metrics/probes bundled and sent to the server together. With your changes, you actually removed all of the metrics in the startup-timeline ping – https://searchfox.org/mozilla-mobile/search?path=%5Efenix&q=startup-timeline – so we can remove the 1) the startup-timeline ping from `pings.yaml` and 2) remove the code that schedules/submits it (i.e. this code right here).

You could either add that change to this PR, add it to a follow-up PR (to keep it simple), or, if you'd rather work on something else, file a follow-up bug to remove it that someone else can work on. Your choice! 😄

**Review by mcomella — COMMENTED** [review]
This looks good to me! @RotBolt Let me know if you'll be addressing the nits or not. If not, I'll approve and merge. If so, I'll await the commits and review the new changes. Thanks! 😀

**Comment by RotBolt** [comment]
I could add the changes in this PR only

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt:80]: Let me know if I got this correct, since I removed `startup-timeline metrics` , so now there is no metrics to send. Now since there no metrics, there is no need to Ping the server for metrics ?
Am I right here ?

Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt]: Update this has been removed

File: app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt:22]: Stat classes has been removed

File: app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt]: Renamed and updated the description as well

**Comment by mcomella** [comment]
Thanks @RotBolt – I overlooked this today so I'll review it tomorrow.

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (mcomella) [reply app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt]: > since I removed startup-timeline metrics , so now there is no metrics to send. Now since there no metrics, there is no need to Ping the server for metrics ?
Am I right here ?

Yep! That's correct!

File: app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt:13]: nitpick (non-blocking): as far as I can tell, the git history/blame for this file was lost. To rename this file, did you use `git mv <old-path> <new-path>` (or an equivalent in the IDE)? If not, in the future, you should use the command for file renames as it preserves the file's history. If so, then it could be that the file was changed so much that git decided to create a new entry for it rather than keep the history

The git commands to fix it now are time consuming and tedious and the file is small so I don't think it's worth our time to address before landing: I just wanted to raise this for the future. :)

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt:34]: nitpick (non-blocking): ideally we'd change this property name too but I don't want to delay us further on another review. Feel free to fix it in a subsequent PR.

File: app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt:17]: thought (non-blocking): in the removed code, there didn't seem to be any tests that test this functionality directly (that applicationInitNanos is set correctly) so the tests were correctly removed. However, this functionality could benefit from a new test. I don't feel so strongly about this that I want to block the PR but just a thought.

**Review by mcomella — APPROVED** [review]
This looks good to me – thanks @RotBolt ! It feels so good to see code get removed. 😁 I'll open a new PR to run CI and merge these changes.

**Comment by mcomella** [comment]
I created https://github.com/mozilla-mobile/fenix/pull/20911 to run the full CI and merge.

File: app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt:17]: I think I could add a test for it in `cleaning unused resource PR`

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt:34]: Yes, I was not sure what to name it. Could add it in `cleaning unused resources PR`

File: app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/perf/ApplicationInitTimeContainer.kt:13]: Will take note of it. I was not aware of it at all

File: app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt
Comment (mcomella) [reply app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt:34]: Usually a good choice is naming it after the class, e.g.:
```kotlin
internal val applicationInitTimeContainer by lazy { ApplicationInitTimeContainer(...) }
```

> Could add it in cleaning unused resources PR

Sounds good. btw, in that "unused resources" PR, please don't remove any unused strings. We have to figure out how unused strings will work with our localization workflow: https://github.com/mozilla-mobile/fenix/issues/20871

**Comment by mcomella** [comment]
btw, I'm trying to land this code in my other PR but I keep running into intermittent test failures unrelated to your changes and unrelated to the errors we ran into before 😓. It'll be landed soon!

**Comment by mcomella** [comment]
The code was merged in https://github.com/mozilla-mobile/fenix/pull/20911 https://github.com/mozilla-mobile/fenix/commit/707918cdc157b682b1ebc7ba7a9f81a5cb38362e so I'm going to close this. Thanks @RotBolt!

**Pull request closed without merging**
